src/scf/potential_mixing.jl

```diff
@@ -47,25 +54,37 @@ function (anderson::AndersonAcceleration)(xₙ, αₙ, Pfxₙ)
         return xₙ .+ αₙ .* Pfxₙ
     end
 
-    M = hcat(Pfxs...) .- vec(Pfxₙ)  # Mᵢⱼ = (Pfxⱼ)ᵢ - (Pfxₙ)ᵢ
+    push!(anderson, xₙ, αₙ, Pfxₙ)
+    #remove large residuals 
+    min_err = minimum(anderson.error)
+    throw_index  = Int[]
+    for i in eachindex(anderson.error)
+        if anderson.δ*anderson.error[i] > min_err #ensure δ|Pfxᵢ| ≤ minᵢ |Pfxᵢ| 
+            push!(throw_index,i)
+        end
+    end
+    deleteat!(anderson,throw_index)
+
+    # Special case where the history is of size 1
+    length(anderson.error) == 1 && return xₙ .+ αₙ .* Pfxₙ
+
+    M =  hcat(Pfxs[2:end]...) - hcat(Pfxs[1:end-1]...)  # Mᵢⱼ = (Pfxⱼ₊₁)ᵢ - (Pfxⱼ)ᵢ
```

Comment:
I wonder how the numerical stability of this compares to our previous approach of using Pfxi - Pfxn. Would be good to test this on a non-trivial case just to check.